### ISSUE TITLE
Vulnerability advisor - split 'name' field.

### ISSUE THREAD

**Issue by lexjacobs** [Solinea/goldstone-server#274]
Split name field into 2 parts.
ossa record, and brief description.

example:
`name: "OSSA-2015-005: Nova console Cross-Site WebSocket hijacking¶"`

Also, please strip `¶` and `↵` characters from return payload.

![image](https://cloud.githubusercontent.com/assets/5633015/12567781/1bb55fc2-c376-11e5-98cf-5bc020da205d.png)

**Comment by bhuvan-rk** [comment]
@lexjacobs  @jxstanford 

 Looks like there is some confusion with the fields in the Vulnerability model. 

There is already an existing ossa-id, name and a description. I'm not sure if we need one more ossa-record field or description fields separately.

In our code, we are truncating the incoming OSSA-ID to display just the numerals as ossa_id, which is not needed. I will fix this one. 

 This is also consistent with how things are in the open-stack web-site. Vulnerabilities are searched by their entire ID string, not just the numerals. 

Also, the name contains its OSSA-ID as well in all forums. So perhaps we should keep it too. 

Convention : 

Name = OSSA_ID + Words/Strings (Keeping it as is)
OSSA_ID = OSSA-DateString ( Ex: OSSA-2016-003 ) and not just date-string alone, I'm fixing this one

**Comment by lexjacobs** [comment]
@bhuvan-rk 

I don't have the vulnerabilities loaded at this moment as I recently dropped and recreated my containers.

To the best of my memory, the part of this discussion that I am aware of included the need for:


* defining a field that will serve as the 'master sort key' to determine what order the server returns records in an unfiltered query. It looks like it should probably be "notice_date". However, that appears to be a string, and not a sortable date field.
* Defining a "slug" for the full text of the vulnerability. The full text seems to be the "description" field. John mentioned a possible source of the slug to be splitting the "name" field on the ":" and using the latter half. The "slug" will be used in the table as a brief description that will invite the user to click on, and expand the full text of the vulnerability.

**Comment by bhuvan-rk** [comment]
@lexjacobs 

I will convert notice-date into a date-time field, so that you can sort and display. 
"slug" will be the name field

Both the name and description fields are pretty neatly written and maintained. So we should probably keep them as-is.

**Comment by lexjacobs** [comment]
@bhuvan-rk 
Okay, I can use the "name" field as the brief description.
And when a user clicks on it, they will get the full "description" field.

Can you also strip those  "¶" characters? They do render in the client. The 'carriage return' symbols are okay as they don't seem to render.

**Comment by lexjacobs** [comment]
Please reassign ticket to Alex for client implementation of updated fields.
Thank you.

**Comment by bhuvan-rk** [comment]
Will remove trailing symbols and re-assign after I check-in the fix.

**Comment by bhuvan-rk** [comment]
Assigning to @lexjacobs for UI part of fix

**Comment by lexjacobs** [comment]
@bhuvan-rk 

errors reported around loading of vulnerabilities.
data failed to populate - client call to `http://localhost:8000/compliance/vulnerability_advisor/openstack/?page_size=10&page=1`
returns `{"count":0,"next":null,"previous":null,"results":[]}`

```
gsappdev_1 | [2016-01-29 17:45:20,079: INFO/Worker-2] Starting new HTTPS connection (1): security.openstack.org
gsappdev_1 | [2016-01-29 17:45:20,537: INFO/Worker-2] goldstone.compliance.tasks.vulnerability_check[24a077f0-5103-4a5d-83ed-5b6924da0155]: Found new vulnerability advisory, id: OSSA-2016-004
gsappdev_1 | [2016-01-29 17:45:20,539: INFO/Worker-2] Starting new HTTPS connection (1): security.openstack.org
gsdb_1     | ERROR:  value too long for type character varying(10)
gsdb_1     | STATEMENT:  INSERT INTO "compliance_vulnerability" ("uuid", "ossa_id", "link", "processed", "name", "description", "affects", "patches", "references", "notes", "cve", "notice_date", "created", "updated") VALUES ('3ab9d527-8f85-42d6-9168-4417473812ef', 'OSSA-2016-004', 'https://security.openstack.org/ossa/OSSA-2016-004.html', false, 'OSSA-2016-004: Swift proxy-server DoS through Large Object¶', 'Romain LE DISEZ from OVH and Örjan Persson from Kiliaro independently
gsdb_1     | 	reported two vulnerabilities in Swift Large Object. By repeatedly
gsdb_1     | 	requesting and interrupting connections to a Large Object (Dynamic or
gsdb_1     | 	Static) URL, a remote attacker may exhausts Swift proxy-server
gsdb_1     | 	resources, potentially resulting in a denial of service. Note that
gsdb_1     | 	there are two distinct bugs that can exhaust proxy resources, one for
gsdb_1     | 	client connection (client to proxy), one for servers connection (proxy
gsdb_1     | 	to server). All Swift setup are affected.', 'Swift: >=2.2.1 <= 2.3.0, >= 2.4.0 <= 2.5.0', 'https://review.openstack.org/270234 (proxy to server) (Kilo)
gsdb_1     | 	https://review.openstack.org/217750 (client to proxy) (Kilo)
gsdb_1     | 	https://review.openstack.org/270235 (proxy to server) (Liberty)
gsdb_1     | 	https://review.openstack.org/270233 (proxy to server) (Mitaka)', 'https://bugs.launchpad.net/bugs/1466549 (client to proxy)
gsdb_1     | 	https://bugs.launchpad.net/bugs/1493303 (proxy to server)
gsdb_1     | 	http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-0737 (client to proxy)
gsdb_1     | 	http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-0738 (proxy to server)', 'The client to proxy issue (CVE-2016-0737) is already fixed in Liberty
gsdb_1     | 	The remaining fix will be included in future 2.3.1 (Kilo) and 2.5.1 (Liberty)
gsdb_1     | 	releases.', 'CVE-2016-0737 (client to proxy), CVE-2016-0738 (proxy to server)', '2016-01-20'::date, '2016-01-29T17:45:20.956147+00:00'::timestamptz, '2016-01-29T17:45:20.956176+00:00'::timestamptz)
gsappdev_1 | [2016-01-29 17:45:20,995: ERROR/MainProcess] Task goldstone.compliance.tasks.vulnerability_check[24a077f0-5103-4a5d-83ed-5b6924da0155] raised unexpected: DataError('value too long for type character varying(10)\n',)
gsappdev_1 | Traceback (most recent call last):
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
gsappdev_1 |     R = retval = fun(*args, **kwargs)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
gsappdev_1 |     return self.run(*args, **kwargs)
gsappdev_1 |   File "/home/app/goldstone/compliance/tasks.py", line 209, in vulnerability_check
gsappdev_1 |     notice_date=notice_date)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/models/manager.py", line 127, in manager_method
gsappdev_1 |     return getattr(self.get_queryset(), name)(*args, **kwargs)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 348, in create
gsappdev_1 |     obj.save(force_insert=True, using=self.db)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/models/base.py", line 734, in save
gsappdev_1 |     force_update=force_update, update_fields=update_fields)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/models/base.py", line 762, in save_base
gsappdev_1 |     updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/models/base.py", line 846, in _save_table
gsappdev_1 |     result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/models/base.py", line 885, in _do_insert
gsappdev_1 |     using=using, raw=raw)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/models/manager.py", line 127, in manager_method
gsappdev_1 |     return getattr(self.get_queryset(), name)(*args, **kwargs)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 920, in _insert
gsappdev_1 |     return query.get_compiler(using=using).execute_sql(return_id)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 974, in execute_sql
gsappdev_1 |     cursor.execute(sql, params)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 79, in execute
gsappdev_1 |     return super(CursorDebugWrapper, self).execute(sql, params)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
gsappdev_1 |     return self.cursor.execute(sql, params)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/utils.py", line 98, in __exit__
gsappdev_1 |     six.reraise(dj_exc_type, dj_exc_value, traceback)
gsappdev_1 |   File "/venv/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
gsappdev_1 |     return self.cursor.execute(sql, params)
gsappdev_1 | DataError: value too long for type character varying(10)
gsappdev_1 |
```

**Comment by lexjacobs** [comment]
@bhuvan-rk confirming drop/recreate of db kicked off migrations and I was able to load new vulnerabilities.

Client changes that affect rendering of update data checked into:
/goldstone-server/goldstone/compliance [issue-274]
and
/goldstone-server [main_issue_274]

`¶` symbols still appearing in `name` field as \u00b6
and explicit carriage (not rendering via client) returns `↵ ` appearing as `\n`.

note: as mentioned before carriage returns don't seem to affect client rendering, but feel free to strip them out.

![image](https://cloud.githubusercontent.com/assets/5633015/12685288/569a3d24-c678-11e5-8c72-8176234ecfea.png)


sample of json being returned:

```
{
    "count": 124,
    "next": "http://localhost:8000/compliance/vulnerability_advisor/openstack/?page=2&page_size=1",
    "previous": null,
    "results": [
{
            "affects": "Heat: <=2015.1.2, ==5.0.0",
            "created": "2016-01-29T19:05:04.677818Z",
            "cve": "CVE-2015-5295",
            "description": "Steven Hardy from Red Hat reported a vulnerability in Heat template\nvalidation. By referencing a local file like /dev/zero, an\nauthenticated user may trick the heat engine service to load arbitrary\nlocal file content resulting in a Denial of Service attack through\nmemory exhaustion. Note that the file content is not written back to\nthe user, though the user can determine if a file exists and if it is\nreadable by heat-engine. All Heat setups are affected.",
            "link": "https://security.openstack.org/ossa/OSSA-2016-003.html",
            "name": "OSSA-2016-003: Heat denial of service through template-validate\u00b6",
            "notes": "This fix will be included in future 2015.1.3 (kilo) and 5.0.1 (liberty)\nreleases.",
            "notice_date": "2016-01-19",
            "ossa_id": "OSSA-2016-003",
            "patches": "https://review.openstack.org/269692 (Kilo)\nhttps://review.openstack.org/269691 (Liberty)\nhttps://review.openstack.org/269689 (Mitaka)",
            "processed": false,
            "references": "https://bugs.launchpad.net/bugs/1496277\nhttp://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2015-5295",
            "updated": "2016-01-29T19:05:04.677832Z",
            "uuid": "0cb1f8d3-25a6-4b2f-afd2-87469c0bd408"
        }
    ]
}
```

**Comment by bhuvan-rk** [comment]
@lexjacobs ok thanks for the heads-up. I'll make one more checkin to remove stray characters in strings.

**Comment by lexjacobs** [comment]
merged  1/29/16 69b8f70c9727c45659070757a7b5efb97707224a
